From: Michal Simek <michal.simek@xilinx.com>
Subject: [U-BOOT PATCH 3/3] microblaze: bootm: Add support for loading
 initrd

fdt_initrd add additional information to DTB about initrd
addresses which are later used by kernel.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 arch/microblaze/lib/bootm.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/microblaze/lib/bootm.c b/arch/microblaze/lib/bootm.c
index 56a1568..fbb43b8 100644
--- a/arch/microblaze/lib/bootm.c
+++ b/arch/microblaze/lib/bootm.c
@@ -68,6 +68,11 @@ int do_bootm_linux(int flag, int argc, char * const argv[],
 
 	if (!of_flat_tree && argc > 3)
 		of_flat_tree = (char *)simple_strtoul(argv[3], NULL, 16);
+
+	/* fixup the initrd now that we know where it should be */
+	if (images->rd_start && images->rd_end && of_flat_tree)
+		fdt_initrd(of_flat_tree, images->rd_start, images->rd_end, 1);
+
 #ifdef DEBUG
 	printf("## Transferring control to Linux (at address 0x%08lx) ",
 	       (ulong)thekernel);
-- 
1.8.2.1


